File: CodinGame/Community/hooch_clash_grouch.py
#!/usr/bin/env python3

from collections import Counter

L,R = map(int,input().split())
V = [X**3 for X in range(L,R+1)]
Cnt = Counter(V[i]+V[j] for i in range(len(V)) for j in range(i,len(V)))
vol_cnt = sum((C>1) for C in Cnt.values())
fun_cnt = sum(C*(C-1) for C in Cnt.values())
print(vol_cnt, fun_cnt)

File: CodinGame/Community/hooch_clash_grouch.rb
#!/usr/bin/env ruby

l,r = gets.split(" ").map(&:to_i)
v = (l..r).map{|x| x**3}
s = v.size-1
w = (0...s).map{|i| (i+1..s).map{|j| v[i]+v[j]}}.flatten.sort.chunk{|x| x}.map{|_,l| l.size}
vol_cnt = w.count{|c| c>1}
fun_cnt = w.map{|c| c*(c-1)}.sum
puts "#{vol_cnt} #{fun_cnt}"
